### PR TITLE
Add logic to discover the authenticated client to be used for subscriptions

### PR DESCRIPTION
**Pull request by tudor-malene**
### Why this change is needed

The wallet extension had no logic to infer which authenticated rpc connection to use for subscription, so it just used the first one.
After this change, it tries to find one based on the content of the topics

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: go/enclave/enclave.go [hunk 1]
```diff
@@ -875,6 +875,8 @@ func (e *enclaveImpl) GetLogs(encryptedParams common.EncryptedParamsGetLogs) (co
 		to = nil
 	}
 
+	// todo - remove BlockHash from the parameters
```
Comment (tudor-malene) [inline go/enclave/enclave.go:878]: unrelated. just a reminder

File: tools/walletextension/accountmanager/account_manager.go [hunk 2]
```diff
@@ -59,20 +61,82 @@ func (m *AccountManager) AddClient(address gethcommon.Address, client *rpc.EncRP
...
+			return nil, fmt.Errorf("could not unmarshal filter criteria from the following JSON: `%s`. Cause: %w", string(filterCriteriaJSON), err)
+		}
+	}
+	for i, topicCondition := range filterCriteria.Topics {
+		if i == 0 {
+			continue
+		}
+		for _, topic := range topicCondition {
+			bitlen := topic.Big().BitLen()
+			if bitlen < 80 || bitlen > 160 {
```
Comment (BedrockSquirrel) [inline tools/walletextension/accountmanager/account_manager.go:111]: This is all a bit magic, probably worth a comment to explain what's being checked here. Maybe even worth pulling out the details of this check into a `doesTopicContainClientAddress()` method or something for readability.

**Review by BedrockSquirrel — COMMENTED** [review]
Minor comments

File: tools/walletextension/accountmanager/account_manager.go [hunk 3]
```diff
@@ -59,20 +61,82 @@ func (m *AccountManager) AddClient(address gethcommon.Address, client *rpc.EncRP
...
+		break
+	}
+
+	// determine the client based on the topics
+	// if none is found use the unauthed client and assume this is a lifecycle method
+	if len(rpcReq.Params) == 0 {
+		return client, nil
+	}
+
+	filterCriteriaJSON, err := json.Marshal(rpcReq.Params[1])
```
[the remark this answers is not in the document]
Comment (BedrockSquirrel) [reply tools/walletextension/accountmanager/account_manager.go:93]: No the check above is for length zero, you're getting the second element, I was worried that there might only be one element. Is this an off-by-one then and you meant to put `Params[0]`?

[the remark this answers is not in the document]
Comment (tudor-malene) [reply tools/walletextension/accountmanager/account_manager.go:93]: oh, yeah

**Review by BedrockSquirrel — APPROVED** [review]
LGTM

**Pull request merged**